tools/cdi-func-test-bad-webserver/main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"crypto/rand"
+	"fmt"
+	"log"
+	"math/big"
+	"net/http"
+	"regexp"
+)
+
+func failHEAD(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "HEAD" {
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+	redirect(w, r)
+}
+
+func flaky(w http.ResponseWriter, r *http.Request) {
+	random, err := rand.Int(rand.Reader, big.NewInt(20))
+	if err != nil {
+		panic(err)
+	}
+	if random.Cmp(big.NewInt(0)) == 0 {
+		// 1-in-20 odds of success
```

Comment:
1 in 20 odds of success, but if I can only attempt 10 tries the allotted time, then the odds of not getting success are pretty high. This will result in a flaky test. I think its better to always return success after X number of tries, where we know X will be within the retry limit of a certain period (crash looping pods have exponential backoff). Also do we want to make sure the first try always fails. With this code there is a 1 in 20 chance the first try will succeed and if the test is expecting a failure that will be a problem.